source/api/commands/invoke.md

```diff
@@ -132,6 +132,29 @@ const double = (n) => n * n
 cy.wrap([reverse, double]).invoke(1, 4).should('eq', 16)
 ```
 
+## Invoking an async function
+
+In this example we invoke a Vuex action that returns a Promise. `cy.invoke` will wait until the Promise resolves after the specified delay which we pass as an argument and only then will continue executing. 
+
+```javascript
+getVuex().invoke('dispatch', 'addTodoAfterDelay', {
+  milliseconds: 2000,
+  title: 'async task'
+})
+
+// log message appears after 2 seconds
+cy.log('after invoke')
+
+// assert UI
+getTodoItems().should('have.length', 1).first().contains('async task')
+```
+
+`getVuex()` in this example is a function that returns the Vuex data store wrapped inside Cypress object to allow E2E tests to access the application state. For that the application saved itself as "window.app".
```

Comment:
I would prefer to have a full example provided. We don't have examples that include hidden code - for example, what is `getVuex`, what is `getTodoItems`?. 

We want everything to be fully copy / pastable to run and if a user copy/pastes this in their code it has zero functionality to them and will error.

I would recommend providing an example of a complete async function that can be included and run by copy/pasting. 

If you want to provide a full recipe in a Vue app, it should go in cypress-example-recipes and linked (as it is at the bottom), but a partial should not be included directly in the docs.

Reply:
Good point 👍